query/stdlib/influxdata/influxdb/rules.go

```diff
@@ -764,3 +767,101 @@ func (PushDownWindowAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (p
 		WindowEvery:       window.Every.Nanoseconds(),
 	}), true, nil
 }
+
+//
+// Push Down of group  aggregates.
+// ReadGroupPhys |> { min, max, mean, count, sum }
+//
+type PushDownGroupAggregateRule struct{}
+
+func (PushDownGroupAggregateRule) Name() string {
+	return "PushDownWindowAggregateRule"
+}
+
+func (rule PushDownGroupAggregateRule) Pattern() plan.Pattern {
+	return plan.OneOf(
+		[]plan.ProcedureKind{
+			universe.MinKind,
+			universe.MaxKind,
+			universe.MeanKind,
+			universe.CountKind,
+			universe.SumKind,
+		},
+		plan.Pat(ReadGroupPhysKind))
+}
+
+func (PushDownGroupAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (plan.Node, bool, error) {
+	// Check the aggregate function spec. Require operation on _value. There
+	// are two feature flags covering count and sum. All the other cases
+	// are specifically turned off.
+	var aggregateMethod string
+	fnNode := pn
+	switch fnNode.Kind() {
+	case universe.MinKind:
+		//	if !feature.PushDownSumAggregateRest().Enabled(ctx) || !caps.HaveMin() {
+		//		return pn, false, nil
+		//	}
+
+		minSpec := fnNode.ProcedureSpec().(*universe.MinProcedureSpec)
+		if minSpec.Column != execute.DefaultValueColLabel {
+			return pn, false, nil
+		}
+		// Not enabled.
+		return pn, false, nil
+	case universe.MaxKind:
+		//	if !feature.PushDownSumAggregateRest().Enabled(ctx) || !caps.HaveMax() {
+		//		return pn, false, nil
+		//	}
+
+		maxSpec := fnNode.ProcedureSpec().(*universe.MaxProcedureSpec)
+		if maxSpec.Column != execute.DefaultValueColLabel {
+			return pn, false, nil
+		}
+		// Not enabled.
+		return pn, false, nil
+	case universe.MeanKind:
+		//	if !feature.PushDownSumAggregateRest().Enabled(ctx) || !caps.HaveMean() {
+		//		return pn, false, nil
+		//	}
+
+		meanSpec := fnNode.ProcedureSpec().(*universe.MeanProcedureSpec)
+		if len(meanSpec.Columns) != 1 || meanSpec.Columns[0] != execute.DefaultValueColLabel {
+			return pn, false, nil
+		}
+		// Not enabled.
+		return pn, false, nil
+	case universe.CountKind:
+		if !feature.PushDownGroupAggregateCount().Enabled(ctx) {
+			return pn, false, nil
+		}
+
+		countSpec := fnNode.ProcedureSpec().(*universe.CountProcedureSpec)
+		if len(countSpec.Columns) != 1 || countSpec.Columns[0] != execute.DefaultValueColLabel {
+			return pn, false, nil
+		}
+		aggregateMethod = "COUNT"
```

Comment:
The storage code is written to test against. My preference would be to clean this up in a separate patch, rather than also touching storage in this patch.

Reply:
I don't get what you are saying. This comment is just about replacing the string literal with some constant we already defined or just `fnNode.Kind()`. I don't know how this became storage code.